packages/runtime-core/src/components/Teleport.ts

```diff
@@ -229,29 +229,26 @@ export const TeleportImpl = {
     parentComponent: ComponentInternalInstance | null,
     parentSuspense: SuspenseBoundary | null,
     optimized: boolean,
-    { um: unmount, o: { remove: hostRemove } }: RendererInternals,
-    doRemove: Boolean
+    { um: unmount, o: { remove: hostRemove } }: RendererInternals
   ) {
-    const { shapeFlag, children, anchor, targetAnchor, target, props } = vnode
+    const { shapeFlag, children, anchor, targetAnchor, target } = vnode
 
     if (target) {
       hostRemove(targetAnchor!)
     }
 
-    // an unmounted teleport should always remove its children if not disabled
-    if (doRemove || !isTeleportDisabled(props)) {
```

Comment:
Maybe should also check if the children contain a Teleport that is not disabled. 
if it does. Then delete the children. is it worth it?